File: ignored-test/csharp-xunit/MyTarget_RetryingTargetWrapperBlockingCloseTest.cs
using Xunit;

namespace None
{
    public class MyTarget
    {
        [Fact(Skip = "")]
        public void RetryingTargetWrapperBlockingCloseTest()
        {
            {
                RetryingIntegrationTest(3, () =>
                {
                    var target = new MyTarget()
                    {
                        ThrowExceptions = 5,
                    };
                    var wrapper = new RetryingTargetWrapper()
                    {
                        WrappedTarget = target,
                        RetryCount = 10,
                        RetryDelayMilliseconds = 5000,
                    };
                    var asyncWrapper = new AsyncTargetWrapper(wrapper)
                    {
                        TimeToSleepBetweenBatches = 1
                    };
                    asyncWrapper.Initialize(null);
                    wrapper.Initialize(null);
                    target.Initialize(null);
                    var exceptions = new List<Exception>();
                    var events = new[]
                    {
                        new LogEventInfo(LogLevel.Debug, "Logger1", "Hello").WithContinuation(exceptions.Add),
                        new LogEventInfo(LogLevel.Info, "Logger1", "Hello").WithContinuation(exceptions.Add),
                        new LogEventInfo(LogLevel.Info, "Logger2", "Hello").WithContinuation(exceptions.Add),
                    };
                    // Attempt to write LogEvents that will take forever to retry
                    asyncWrapper.WriteAsyncLogEvents(events);
                    // Wait a little for the AsyncWrapper to start writing
                    System.Threading.Thread.Sleep(50);
                    // Close down the AsyncWrapper while busy writing
                    asyncWrapper.Close();
                    // Close down the RetryingWrapper while busy retrying
                    wrapper.Close();
                    // Close down the actual target while busy writing
                    target.Close();
                    // Wait a little for the RetryingWrapper to detect that it has been closed down
                    System.Threading.Thread.Sleep(200);
                    // The premature abort, causes the exception to be logged
                    Assert.NotNull(exceptions[0]);
                });
            }
        }
    }
}

File: ignored-test/java-junit/MyTarget_RetryingTargetWrapperBlockingCloseTest.java
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Disabled;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import java.util.ArrayList;
import java.util.List;

public class MyTarget {

    @Test
    @Disabled("")
    public void retryingTargetWrapperBlockingCloseTest() throws Exception {
        retryingIntegrationTest(3, () -> {
            MyTarget target = new MyTarget();
            target.setThrowExceptions(5);

            RetryingTargetWrapper wrapper = new RetryingTargetWrapper();
            wrapper.setWrappedTarget(target);
            wrapper.setRetryCount(10);
            wrapper.setRetryDelayMilliseconds(5000);

            AsyncTargetWrapper asyncWrapper = new AsyncTargetWrapper(wrapper);
            asyncWrapper.setTimeToSleepBetweenBatches(1);

            asyncWrapper.initialize(null);
            wrapper.initialize(null);
            target.initialize(null);

            List<Exception> exceptions = new ArrayList<>();
            LogEventInfo[] events = new LogEventInfo[] {
                new LogEventInfo(LogLevel.DEBUG, "Logger1", "Hello").withContinuation(exceptions::add),
                new LogEventInfo(LogLevel.INFO, "Logger1", "Hello").withContinuation(exceptions::add),
                new LogEventInfo(LogLevel.INFO, "Logger2", "Hello").withContinuation(exceptions::add),
            };

            asyncWrapper.writeAsyncLogEvents(events);
            Thread.sleep(50);
            asyncWrapper.close();
            wrapper.close();
            target.close();
            Thread.sleep(200);

            assertNotNull(exceptions.get(0));
        });
    }
}
